src/__init__.py

```diff
@@ -121,22 +133,14 @@ def run(base_path: Path, discourse: Discourse, user_inputs: UserInputs) -> dict[
     """
     metadata = get_metadata(base_path)
     has_docs_dir = has_docs_directory(base_path=base_path)
+    repository = create_repository_client(
+        access_token=user_inputs.github_access_token, base_path=base_path
+    )
+    clients = Clients(discourse=discourse, repository=repository)
```

Comment:
What's the reason behind wrapping both clients in the `Client` object?

Reply:
To reduce the number of arguments that need to be passed around, there were many cases of having both discourse and repository as an argument, this simplifies it down to 1 argument

Reply:
I find it a little obscure, TBH, but I get the point